client/rest-high-level/src/main/java/org/elasticsearch/client/security/PutUserRequest.java

```diff
@@ -149,18 +115,15 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 Arrays.fill(charBytes, (byte) 0);
             }
         }
-        if (roles != null) {
-            builder.field("roles", roles);
-        }
-        if (fullName != null) {
-            builder.field("full_name", fullName);
-        }
-        if (email != null) {
-            builder.field("email", email);
+        builder.field("roles", user.getRoles());
+        if (user.getFullName() != null) {
+            builder.field("full_name", user.getFullName());
         }
-        if (metadata != null) {
-            builder.field("metadata", metadata);
+        if (user.getEmail() != null) {
+            builder.field("email", user.getEmail());
         }
+        builder.field("metadata", user.getMetadata());
+        builder.field("enabled", enabled);
```

Comment:
`enabled` has to be part of the request entity, otherwise all users are put/updated as enabled.